src/pages/Task.tsx

```diff
@@ -76,11 +76,13 @@ export const Task: React.FC = () => {
   const isPatchOnCommitQueue = requester === commitQueueRequester;
 
   // Set the execution if it isnt provided
-  if (Number.isNaN(selectedExecution) && latestExecution !== undefined) {
-    updateQueryParams({
-      execution: `${latestExecution}`,
-    });
-  }
+  useEffect(() => {
+    if (Number.isNaN(selectedExecution) && latestExecution !== undefined) {
+      updateQueryParams({
+        execution: `${latestExecution}`,
+      });
+    }
+  }, [task]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Is it not sufficient to include `selectedExecution` and `latestExecution` here? This would cause the hook to fire again whenever any field within the `task` updates.

Reply:
ahh true!! lemme give it a try P: